service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapper.java

```diff
@@ -264,6 +252,55 @@ private String prepareNarrativeStatement(Observation observation, String comment
         return TemplateUtils.fillTemplate(NARRATIVE_STATEMENT_TEMPLATE, narrativeStatementTemplateParameters.build());
     }
 
+    private String prepareStatementsForDerivedObservations(IdMapper idMapper, List<Observation> derivedObservations) {
+        StringBuilder derivedObservationsBlock = new StringBuilder();
+
+        derivedObservations.forEach(derivedObservation -> {
+            var observationStatement = prepareObservationStatement(idMapper, derivedObservation);
+            var narrativeStatement = prepareNarrativeStatement(idMapper, derivedObservation);
+
+            if (!observationStatement.isEmpty() && !narrativeStatement.isEmpty()) {
```

Comment:
Should we also check for blankness?

Reply:
It should only ever be empty. But wouldn't hurt checking for blankness since it shouldn't be outputted then obviously